internal/services/batch/batch_pool_resource.go

```diff
@@ -603,25 +853,6 @@ func resourceBatchPoolCreate(d *pluginsdk.ResourceData, meta interface{}) error
 
 	parameters.PoolProperties.ScaleSettings = scaleSettings
 
-	nodeAgentSkuID := d.Get("node_agent_sku_id").(string)
-
-	storageImageReferenceSet := d.Get("storage_image_reference").([]interface{})
-	imageReference, err := ExpandBatchPoolImageReference(storageImageReferenceSet)
-	if err != nil {
-		return fmt.Errorf("creating %s: %+v", id, err)
-	}
-
-	if imageReference != nil {
-		// if an image reference ID is specified, the user wants use a custom image. This property is mutually exclusive with other properties.
-		if imageReference.ID != nil && (imageReference.Offer != nil || imageReference.Publisher != nil || imageReference.Sku != nil || imageReference.Version != nil) {
-			return fmt.Errorf("creating %s: Properties version, offer, publish cannot be defined when using a custom image id", id)
-		} else if imageReference.ID == nil && (imageReference.Offer == nil || imageReference.Publisher == nil || imageReference.Sku == nil || imageReference.Version == nil) {
-			return fmt.Errorf("creating %s: Properties version, offer, publish and sku are mandatory when not using a custom image", id)
-		}
-	} else {
-		return fmt.Errorf("creating %s: image reference property can not be empty", id)
-	}
-
```

Comment:
This check logic is moved to expandBatchPoolVirtualMachineConfiguration in batch_pool.go